sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BatchLoads.java

```diff
@@ -77,12 +91,23 @@
     this.singletonTable = singletonTable;
     this.dynamicDestinations = dynamicDestinations;
     this.destinationCoder = destinationCoder;
+    this.maxNumWriters = DEFAULT_MAX_NUM_WRITERS;
   }
 
   void setTestServices(BigQueryServices bigQueryServices) {
     this.bigQueryServices = bigQueryServices;
   }
 
+  /** Get the maximum number of file writers that will be open simultaneously in a bundle. */
+  public int getMaxNumWriters() {
+    return maxNumWriters;
+  }
+
+  /** Set the maximum number of file writers that will be open simultaneously in a bundle. */
+  public void setMaxNumWriters(int maxNumWriters) {
```

Comment:
Should probably be called setMaxNumWritersPerBundle.

Also there should be some caveats about the fact that multiple bundles may be running concurrently, if this is a multi-core worker...